### ISSUE TITLE
Battery percent entity 'stuck' on one of my three devices

### ISSUE THREAD

**Issue by AnthonyBe** [sebr/bhyve-home-assistant#236]
**Describe the bug**
I've been using the integration without issue for many months. After changing the batteries on one of my devices recently, I noticed that the battery percent reading in HA got 'stuck' on 90.83333333%

I've disabled and re-enabled the device in the integration config.
Powered the BHyve device on and off numerous times.
Nothing seems to rectify the issue.
I have 2 other BHyve devices and they are reporting battery percent without issue.
The BHyve device I'm seeing the issue with, is correctly reporting all other attirbutes, only battery percent is causing me an issue.
The battery percent is showing correctly in the BHyve app (i.e. not a deice/Cloud/App/Platform issue)
The battery percent is showing correctly in the attached integration diagnostic data! So I'm REALLY confused by this! It is as though the battery level sensor entity is 'stuck'!
The only thing I haven't tried yet is deleting the integration and re-installing.

![image](https://github.com/sebr/bhyve-home-assistant/assets/17214552/e3ca27c8-fdd6-478c-887a-4f9c498e5a7a)
![image](https://github.com/sebr/bhyve-home-assistant/assets/17214552/1dec932d-6696-4d64-9a7e-760610f7e9ba)

**Expected behaviour**
Correct battery percent values displayed

**BHyve devices**
1 x BHyve Hub
3 x BHyve tap timers

**Device Diagnostics**

[config_entry-bhyve-5bc4b4819e8722eb2f3b99b8df98a730.json](https://github.com/sebr/bhyve-home-assistant/files/15055776/config_entry-bhyve-5bc4b4819e8722eb2f3b99b8df98a730.json)

In addition, I see this error in my HA log:

```
2024-04-22 09:33:02.043 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.back_garden_battery_level fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 951, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1268, in async_device_update
    await self.async_update()
  File "/config/custom_components/bhyve/sensor.py", line 155, in async_update
    await super().async_update()
  File "/config/custom_components/bhyve/__init__.py", line 247, in async_update
    self._on_ws_data(ws_event)
  File "/config/custom_components/bhyve/sensor.py", line 148, in _on_ws_data
    battery_level = self.parse_battery_level(event)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/bhyve/sensor.py", line 175, in parse_battery_level
    battery_level = battery_data.get("percent", 0)
                    ^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'get'
```

**Comment by AnthonyBe** [comment]
Just updated to 3.2.3 and it's resolved the issue. Thanks @sebr !

**Comment by jhemak** [comment]
Started seeing this today, for the first time, after updating:

`WARNING (MainThread) [custom_components.bhyve.sensor] Unexpected battery data, returning 0: battery_status`

**Comment by sebr** [comment]
@jhemak thanks - fixed in [v3.2.4](https://github.com/sebr/bhyve-home-assistant/releases/tag/3.2.4)

**Comment by jhemak** [comment]
Thank you!  Seems resolved for me now.
